pkg/dataobj/metastore/iter.go

```diff
@@ -123,60 +121,68 @@ func forEachStreamSectionPointer(
 
 				switch pointerCol.Type {
 				case pointers.ColumnTypePath:
+					values := col.(*array.String)
```

Comment:
here and below - small improvement to convert the column only once (instead of doing it for every row)